courses/advanced-foundry/6-account-abstraction/1-introduction/+page.md

```diff
@@ -0,0 +1,78 @@
+
+### Ethereum – EntryPoint.sol
+Ethereum implements account abstraction using a smart contract called `EntryPoint.sol`. This contract acts as a gateway for handling user operations and transactions in a more flexible manner.
+
+If you're interested, you can checkout the code here: [EntryPoint.sol on GitHub](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/core/EntryPoint.sol)
+
+
+### zkSync – Natively Integrated
+zkSync, on the other hand, has account abstraction natively integrated into its codebase. This allows for seamless handling of transactions and operations without the need for additional contracts.
+
+## Account Abstraction Uses Alt-Mempools
+
+![Current Wallet Issues](/6-account-abstraction/1-introduction/userOp.png)
+
+### User Operations (Off-Chain)
+In Ethereum, user operations are first sent off-chain. This means that the initial handling and validation occur outside the main blockchain network, reducing congestion and improving efficiency. In the above example, the user operation is signed with Google and is sent to the alt-mempool, which then sends it to the main blockchain network.
+
+### Transactions and Gas Payments (On-Chain)
+Once validated, the user operations are sent on-chain as transactions. These transactions are executed and gas fees are paid on behalf of the user, directly from their account. This is managed through the `EntryPoint.sol` contract. From here, the user's smart contract essentially becomes their wallet. If gas has not yet been settled, it is paid for by the user's account. Finally, the contract is deployed to the blockchain. 
+
+### EntryPoint.sol Optional Add-ons
+The `EntryPoint.sol` contract also allows for optional add-ons, such as a Signature Aggregator and a Pay Master. These add-ons can be used to further optimize gas fees and improve user experience.
+
+#### Signature Aggregator
+An optional add-on to the `EntryPoint.sol` contract is the signature aggregator. This component collects and verifies multiple signatures, ensuring that only authorized transactions are processed.
+
+#### Pay Master
+Another optional component is the pay master. It handles gas payments, allowing users to pay for transactions in various ways, not limited to the native cryptocurrency.
+
+## zkSync
+
+![zkSync Enrty Point](/6-account-abstraction/1-introduction/zksyncEntryPoint.png)
+
+### Acts as an Alt-Mempool
+In zkSync, the account abstraction mechanism acts similarly to an alt-mempool. It efficiently manages user operations and transactions, ensuring that they are processed securely and promptly.
+
+By understanding these concepts, you'll have a solid foundation in account abstraction and its implementation in leading blockchain platforms like Ethereum and zkSync. 
+
+## Questions for Review
+
+Before we move one, here are some questions to help you review what we've covered so far **(Watch Video Lesson firts!)**:
```

Comment:
Let's remove this phrase: (Watch Video Lesson first!)
as the written lessons are meant to provide the same information as the video lessons:)